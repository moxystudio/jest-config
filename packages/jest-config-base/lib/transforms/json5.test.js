'use strict';

const fs = require('fs');
const tmp = require('tmp-sync');
const json5Transformer = require('./json5');

const dir = tmp.in(__dirname);

afterAll(() => {
    tmp.clean();
});

it('should return JSON of the file', () => {
    const contents = `
// This is a comment
{
    "hello": "world",
}
`;
    const filename = `${dir}/test-file.json5`;

    fs.writeFileSync(filename, contents);

    const output = json5Transformer.process(contents, filename, {});

    expect(output).toMatchSnapshot();
});
