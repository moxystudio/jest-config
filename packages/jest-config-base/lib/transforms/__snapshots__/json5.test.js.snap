// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return JSON of the file 1`] = `"module.exports = {\\"hello\\":\\"world\\"};"`;
